src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,117 +1,79 @@
 import clsx from 'clsx';
-import React, {
-  useRef,
-  useEffect,
-  ChangeEventHandler,
-  MutableRefObject,
-} from 'react';
+import React from 'react';
+import { useUID } from 'react-uid';
 import styles from './Checkbox.module.css';
-import { ENTER_KEYCODE } from '../../util/keycodes';
+import CheckboxInput from '../CheckboxInput';
+import type { CheckboxInputProps } from '../CheckboxInput';
+import CheckboxLabel from '../CheckboxLabel';
+import type { CheckboxLabelProps } from '../CheckboxLabel';
 
-export interface Props {
+// id is required in CheckboxInputProps but optional in CheckboxProps, so we
+// first remove `id` from CheckboxInputProps before intersecting.
+export type CheckboxProps = Omit<CheckboxInputProps, 'id'> & {
   /**
-   * HTML id of the helper text used to describe the component
-   */
-  ariaDescribedBy?: string;
-  /**
-   * Toggles whether or not the checkbox is checked or unchecked
-   */
-  checked?: boolean;
-  /**
-   * CSS class names that can be appended to the component.
-   */
-  className?: string;
-  /**
-   * Disables the field and prevents editing the contents
-   */
-  disabled?: boolean;
-  /**
-   * HTML id for the component
+   * HTML id attribute. If not passed, this component
+   * will generate an id to use for accessibility.
    */
   id?: string;
   /**
-   * Indeterminate prop that turns the checkbox indeterminate state on
-   */
-  indeterminate?: boolean;
-  /**
-   * Inverted variation for dark backgrounds
-   */
-  inverted?: boolean;
-  /**
-   * HTML name attribute for the checkbox
-   */
-  name?: string;
-  /**
-   * Function that fires when field value has changed
+   * Visible text label for the checkbox.
    */
-  onChange?: ChangeEventHandler;
+  label?: React.ReactNode;
   /**
-   * Toggles the form control's interactivity. When `readOnly` is set to `true`, the form control is not interactive
+   * Size of the checkbox label.
    */
-  readOnly?: boolean;
-  /**
-   * A string representing the value of the checkbox
-   */
-  value?: string;
-}
+  size?: CheckboxLabelProps['size'];
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Checkbox} from "@chanzuckerberg/eds-components";
+ * ```
+ * ```ts
+ * import Checkbox, {CheckboxInput, Label} from '@chanzuckerberg/eds-components/lib/Checkbox';
```

Comment:
```suggestion
 * import {CheckboxInput, Label} from '@chanzuckerberg/eds-components';
```

also added to Breaking changes doc

Reply:
Thanks for adding to the breaking changes doc!